lib/ansible/plugins/action/synchronize.py

```diff
@@ -155,6 +155,11 @@ def run(self, tmp=None, task_vars=None):
         if task_vars is None:
             task_vars = dict()
 
+        # We make a copy of the args here because we may fail and be asked to
+        # retry. If that happens we don't want to pass the munged args through
+        # to our next invocation. Munged args are single use only.
+        self._task._tmp_args = self._task.args.copy()
```

Comment:
The problem as I understand it is this _task object is reused on every retry. So while it may be a copy the first time around it isn't a fresh copy each time.